common/raw_hashtable.h

```diff
@@ -1399,79 +1503,95 @@ BaseImpl<InputKeyT, InputValueT, InputKeyContextT>::GrowAndInsert(
 template <typename InputBaseT, ssize_t SmallSize>
 TableImpl<InputBaseT, SmallSize>::TableImpl(const TableImpl& arg)
     : BaseT(arg.alloc_size(), arg.growth_budget_, SmallSize) {
+  // Check for completely broken objects. These invariants should be true even
+  // in a moved-from state.
+  CARBON_DCHECK(arg.alloc_size() == 0 || !arg.is_small() ||
+                arg.alloc_size() == SmallSize);
   CARBON_DCHECK(arg.small_alloc_size_ == SmallSize);
+  CARBON_DCHECK(this->small_alloc_size_ == SmallSize);
 
-  ssize_t local_size = arg.alloc_size();
+  SetUpStorage();
+  this->CopySlotsFrom(arg);
```

Comment:
```suggestion
  if (this->alloc_size() != 0) {
    SetUpStorage();
    this->CopySlotsFrom(arg);
  }
```
I think we want copying from a moved-from object to not allocate / copy.

Reply:
I'm... not sure how this change got lost. Sorry. I swear I had this exact code in there last time I looked. =]